### DISCUSSION TITLE
Decision reponsiveness

### DISCUSSION THREAD

**Discussion by nfrgosselin** [gitcoinco/grants-stack#335] [Q&A]
Hey team — I've noticed that there's still a lag time from approval transaction completing to the app updating the card status. You also need to refresh the page in order to see the decision go through: https://share.vidyard.com/watch/fhi2sEths1KdFrSRNbtqGU?

Are we confident that the fixes from the Graph will solve this ([conversation in Discord here](https://discord.com/channels/562828676480237578/1011646812127502427)) ? Are there other things we should consider?

**Comment by thelostone-mc** [comment]
Decisioning no longer relies on subgraph but instead fetched from contracts directly. 
The only scenario where there is a delay if the transaction is pending. Once txn is passed , the decision should auto reflect. 

I beleive the issue is that the UI doesn't get refreshed automatically when the txn passed which is what gives the false impression that it's taking a long time 


Could you try this once more and refresh your screen immediately after the txn passes ? 
It should be near instant

**Comment by nfrgosselin** [comment]
I thought it might be a refresh issue too, but I'm still seeing a delay between the tx confirmation and UI update. I just rejected my application on this round and it took about a minute for the the update to go through (even after several refreshes)

![image](https://user-images.githubusercontent.com/15041635/188500197-bfaf7108-c31f-430c-8776-3f7c45700097.png)

**Comment by thelostone-mc** [comment]
Yup i'm willing to bet that it's solely the refresh issue. Might want to create a ticket for that -> we could do something like the frontend keeps checking the txn status until it's passed and then refresh screen

**Reply by nfrgosselin** [reply]
I'm confused — if I'm seeing the transaction complete in Metamask but I'm not seeing the decision after (multiple) refresh, how is that a refresh issue?

**Reply by shavinac** [reply]
i think we are still relying on subgraph to update with the latest `projectsMetaPtr` value from onchain, so there might still be a few min delay while subgraph re-indexes.

**Comment by thelostone-mc** [comment]
@nfrgosselin One thing which we could explore is adding a waiting state.

Once the txn is signed. 
In that session -> we attempt to fetch the info from subgraph and until that information is available -> we show a indexing by subgrpah piece as well 

I feel this could be part of the progress modal which we already have and this would be applicable in 

- creating a program / round
- approving / rejecting projects in a round 

(Bascially any on-chain interaction piece)

**Reply by melissa-neira** [reply]
@thelostone-mc Here's a possible solution, the copy can be changed. 

![Processing](https://user-images.githubusercontent.com/87520659/189379184-b92fd790-2248-44e1-9032-3cc15ae7afab.png)

**Reply by nfrgosselin** [reply]
I'm cautiously aligned with this — is there any indication that the subgraph indexing speed will improve? I'm worried about someone being stuck in this state for a long time

**Reply by thelostone-mc** [reply]
> Is there any indication that the subgraph indexing speed will improve? 

The indexing speed would not improve but at the same time it would slow down. 
Having this state at least makes it clear that there is a in progress step (so that might be less confusing to the user ? )

It's like the processing payment page / waiting for bank to say the charge has been made

**Reply by thelostone-mc** [reply]
Defer to you on this @nfrgosselin / @melissa-neira though  🙌🏼

**Comment by vacekj** [comment]
Closing as stale.
